upup/pkg/fi/cloudup/awstasks/route.go

```diff
@@ -32,9 +32,13 @@ type Route struct {
 	Name *string
 
 	RouteTable      *RouteTable
-	InternetGateway *InternetGateway
 	Instance        *Instance
 	CIDR            *string
+
+	// Either an InternetGateway or a NAT Gateway
+	// MUST be provided.
+	InternetGateway *InternetGateway
+	NatGateway      *NatGateway
 }
 
 func (e *Route) Find(c *fi.Context) (*Route, error) {
```

Comment:
> What the heck are we doin here

We are attempting to find a resource via the AWS API.

That function name is find to me - but we could probably use some documentation around what the Find() method in this package is expected to do..

Changing these would be a huge refactor IMHO